__tests__/organizations.ts

```diff
@@ -413,6 +425,258 @@ describe('mutation updateOrganization', () => {
   });
 });
 
+describe('mutation joinOrganization', () => {
+  const MUTATION = /* GraphQL */ `
+    mutation JoinOrganization($id: ID!, $token: String!) {
+      joinOrganization(id: $id, token: $token) {
+        role
+        referralToken
+        organization {
+          id
+          name
+        }
+      }
+    }
+  `;
+
+  beforeEach(async () => {
+    await con.getRepository(Feed).save([
+      {
+        id: '1',
+        userId: '1',
+      },
+      {
+        id: '2',
+        userId: '2',
+      },
+      {
+        id: '3',
+        userId: '3',
+      },
+    ]);
+
+    await con.getRepository(ContentPreferenceOrganization).save([
+      {
+        userId: '1',
+        referenceId: 'org-1',
+        organizationId: 'org-1',
+        feedId: '1',
+        status: ContentPreferenceStatus.Follow,
+        flags: {
+          role: OrganizationMemberRole.Owner,
+          referralToken: 'ref-token-1',
+        },
+      },
+      {
+        userId: '2',
+        referenceId: 'org-1',
+        organizationId: 'org-1',
+        feedId: '2',
+        status: ContentPreferenceStatus.Follow,
+        flags: {
+          role: OrganizationMemberRole.Member,
+          referralToken: 'ref-token-2',
+        },
+      },
+    ]);
+  });
+
+  it('should not authorize when not logged-in', () =>
+    testMutationErrorCode(
+      client,
+      { mutation: MUTATION, variables: { id: 'org-1', token: 'ref-token-1' } },
+      'UNAUTHENTICATED',
+    ));
+
+  it('should return forbidden if token and organization do not match', async () => {
+    loggedUser = '3';
+
+    testMutationErrorCode(
+      client,
+      {
+        mutation: MUTATION,
+        variables: { id: 'non-existing', token: 'ref-token-1' },
+      },
+      'FORBIDDEN',
+      'Invalid invitation token',
+    );
+  });
```

Comment:
Maybe also another case where the org exists just to be sure.

Reply:
Added